packages/web/src/components/Header/README.md

```diff
@@ -0,0 +1,272 @@
+</header>
+```
+
+## Color Variants
+
+Currently, Header comes in two color variants: **transparent** (for dark
+backgrounds) and **inverted** (for light backgrounds). Add `Header--inverted`
+modifier class to make Header background inverted.
+
+```html
+<header class="Header Header--inverted">
+  <a href="#">
+    <img src="…" alt="Logo" />
+  </a>
+</header>
+```
+
+## Simple Header
+
+The `Header--simple` modifier makes the header bar slightly shorter and aligns
+its content to center. Use this design variant when all you need on the page is
+just branding.
+
+```html
+<header class="Header Header--simple">
+  <a href="#">
+    <img src="…" alt="Logo" />
+  </a>
+</header>
+```
+
+## Supported Content
+
+In order to create a responsive header with top-level navigation, there are the
+following building blocks to use:
+
+1. toggle button,
+2. actions bar with primary and secondary action slots,
+3. navigation,
+4. desktop dialog.
+
+### Toggle Button
+
+Toggle button reveals actions on small screens (up to the desktop breakpoint).
+It makes use of inverted Button by default.
+
+```html
+<div class="Header__toggle">
+  <button
+    type="button"
+    class="Button Button--inverted"
+    data-toggle="header"
+    data-target="#my-header-actions"
+    aria-controls="my-header-actions"
+    aria-expanded="false"
+  >
+    <span class="Header__icon Header__icon--menu" aria-hidden="true"></span>
+    <span class="accessibility-hidden">Menu</span>
+  </button>
+</div>
+```
+
+### Actions Bar
+
+Actions bar is a wrapper that holds one slot for primary actions (aligned to
+left), one slot for secondary actions (aligned to right), and a closing button.
+On small screens, actions are presented in form of an off-canvas pane sliding
+from right. From the desktop breakpoint up, the actions bar is rendered inside
+Header.
+
+👉 While vertical overflow is correctly handled on small screens
+(`Header__content` provides vertical scrolling automatically), it is critical to
+make sure all your actions fit the Header on the desktop breakpoint.
+
+```html
+<nav id="my-header-actions" class="Header__bar">
+  <div class="Header__content">
+    <button
+      type="button"
+      class="Header__close"
+      data-toggle="header"
+      data-target="#my-header-actions"
+      aria-controls="my-header-actions"
+      aria-expanded="false"
+    >
+      <span class="Header__icon Header__icon--close" aria-hidden="true"></span>
+      <span class="accessibility-hidden">Close</span>
+    </button>
+    <div class="Header__actions Header__actions--primary">
+      <!-- Primary actions -->
+    </div>
+    <div class="Header__actions Header__actions--secondary">
+      <!-- Secondary actions -->
+    </div>
+  </div>
+</nav>
+```
+
+### Navigation
+
+Navigation is designed to live in either of the action slots. As of now, only
+single-level navigation is supported. You may consider using the desktop dialog
+for special use cases though.
+
+```html
+<ul class="Header__nav">
```

Comment:
I think it is too connected to split it.

Reply:
There has been some consideration before naming things here the way they are so I will try my best to explain 🙂.

---

`Nav` and `NavItem` would suggest they work also outside of `Header` which is not true.

I also investigated if the SCSS could be split into multiple files (`Header.scss` has become quite long) but it turned out the styles are too tightly connected together (which makes the output CSS as lightweight and DRY as possible).

To be more specific, this: 
![obrazek](https://user-images.githubusercontent.com/5614085/162258564-16192fd5-52d5-4466-a0fc-25a1cb66579a.png)

… is just another appearance of the same HTML:

![obrazek](https://user-images.githubusercontent.com/5614085/162258705-10fb2aea-58a2-4cb1-a946-f023e020311d.png)

Also the entire off-canvas pane on small screens:

![obrazek](https://user-images.githubusercontent.com/5614085/162258864-5b4b923e-ea74-4eb9-9077-ebffa673a6cd.png)

… shares the same HTML with a different styling on desktop:

![obrazek](https://user-images.githubusercontent.com/5614085/162258971-1ac77091-3a6e-4d26-b65f-569f89a2796f.png)

These things **are not different components**. They have the same HTML which happens to have radically different styling below and above desktop breakpoint. (Sharing the same HTML is a good thing for SEO, DOM size and also for maintenance, that's why.)

So these parts cannot be pulled out of `Header` because they would not work anywhere else—they are tailor-made for the context of `Header`. Hence I believe the `Header` namespace (or prefix) is the right thing for them.

---

Contrarily, the only component that turned out to be able to live outside of `Header` is current `Header Dialog` (click _Marian_ on a 1280+ screen to open it):

![obrazek](https://user-images.githubusercontent.com/5614085/162259258-f46f3097-8f66-41b8-9f27-133d65dd529b.png)

It shares the same CSS with navigation on previous pictures, but it has its own HTML which is not shared with anything in `Header`. While we don't need it to be usable on its own outside of `Header` at the time being, I can imagine it could become independent in future, e.g. as `OffCanvas` component or similar. At that point (and no sooner) we may also need to have some navigation component that could live inside it. Maybe that will be the time to pull out `Nav`, `NavItem` and `NavLink` out of `Header`, maybe not.

---

So yes, the CSS is too interconnected to be split now as @crishpeen states, and I don't think it's wrong because it exactly reflects current design requirements which we don't expect to change in the near future.